fedimint-client/src/module/recovery.rs

```diff
@@ -102,3 +102,51 @@ impl IntoDynInstance for NoModuleBackup {
         DynModuleBackup::from_typed(instance_id, self)
     }
 }
+
+/// Progress of the recovery
+///
+/// This includes "magic" value: if `total` is `0` the progress is "not started
+/// yet"/"empty"/"none"
+#[derive(Debug, Copy, Clone, Encodable, Decodable)]
+pub struct RecoveryProgress {
+    pub complete: u32,
+    pub total: u32,
+}
+
+impl RecoveryProgress {
```

Comment:
Maybe a `fn to_percent(&self) -> f32` would also be useful.

Reply:
numerator + denominator seem nice and informative, and the end application can convert as they please.